test/e2e/nfs.go

```diff
@@ -0,0 +1,225 @@
+/*
+Copyright 2015 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+ * This test checks that NFS volume is working. It uses one pod exporting a
+ * directory and other pod consuming it, exporting its content through http.
+ *
+ * To export something via NFS, the pod must be privileged (it mounts
+ * /proc/fs/nfsd). The test reports 'success' on clusters with
+ * --allow_privileged=False (which is the default).
+ */
+
+package e2e
+
+import (
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("NFS", func() {
+	clean := true // If 'false', the test won't clear its namespace (and pods and services) upon completion. Useful for debugging.
+
+	// filled in BeforeEach
+	var c *client.Client
+	var namespace string
+
+	BeforeEach(func() {
+		var err error
+		c, err = loadClient()
+		Expect(err).NotTo(HaveOccurred())
+		namespace = "e2e-ns-" + dateStamp() + "-0"
+	})
+
+	AfterEach(func() {
+		if !clean {
+			return
+		}
+		c.Namespaces().Delete(namespace)
+	})
+
+	It("should be able to mount NFS", func() {
+		podClient := c.Pods(namespace)
+		serviceClient := c.Services(namespace)
+
+		By("creating a NFS server pod")
+		nfs_pod := &api.Pod{
+			TypeMeta: api.TypeMeta{
+				Kind:       "Pod",
+				APIVersion: "v1beta3",
+			},
+			ObjectMeta: api.ObjectMeta{
+				Name: "nfs-server-" + string(util.NewUUID()),
+				Labels: map[string]string{
+					"role": "nfs-server",
+				},
+			},
+			Spec: api.PodSpec{
+				Containers: []api.Container{
+					{
+						Name:       "nfs-server",
+						Image:      "jsafrane/nfs-data",
```

Comment:
In general we don't want to depend on random containers (no offense), especially when they are 400+ MB.  We should make a dir in contrib/for-demos to host the Dockerfile and associated scripts, so that the kubernetes team owns it and can push updates at will.

Then we really need a stripped down container - we can't have every e2e run pulling down hundreds and hundreds of MB.

@zml @satnam6502 @quinton-hoole Do we have a general policy on this?  It's one thing for a random example to use random containers, it's another thing entirely to put it in e2e.